### PR TITLE
[16.0] edi_storage_oca: replace use of 'fs_storage' deprecated functions

### PR DESCRIPTION
**Pull request by sebalix**
Tested with SFTP protocol.

cc @simahawk

### CODE REVIEW

**Review by simahawk — COMMENTED** [review]
what about `find_files`? See #64

**Comment by sebalix** [comment]
I'm suspecting (but I'm not sure) it was the use of these deprecated methods that triggers connection issues over time. These methods are decorated, and it could be that the decorator is keeping an old reference to `self.__fs` having a socket that gets closed after some time. That could explain why the "Test connection" button of `fs_storage` provides the expected result, while the EDI jobs are failing even when requeued.

This will be tested anyway.

File: edi_storage_oca/utils.py [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+import base64
+import os
+import re
+
+
+def add_file(storage, path, filedata, binary=False):
+    data = filedata
+    if not binary:
+        data = base64.b64decode(filedata)
+    relative_path = path.as_posix()
```
Comment (JordiMForgeFlow) [inline edi_storage_oca/utils.py]: we should use `PurePath(path).as_posix()`

File: edi_storage_oca/utils.py [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+    regex = re.compile(pattern)
+    for file_path in fs.ls(relative_path, detail=False):
+        # fs.ls returns a relative path
+        if regex.match(os.path.basename(file_path)):
+            result.append(file_path)
+    return result
+
+
+def get_file(storage, path, binary=False):
+    relative_path = path.as_posix()
```
Comment (JordiMForgeFlow) [inline edi_storage_oca/utils.py]: same issue as in the _add_file_ method

File: edi_storage_oca/components/receive.py [hunk 3]
```diff
@@ -14,7 +14,4 @@ class EDIStorageReceiveComponent(Component):
     _usage = "storage.receive"
 
     def receive(self):
-        path = self._get_remote_file_path("pending")
-        # TODO: clean this up, .get is deprecated in fs_storage
-        filedata = self.storage.get(path.as_posix())
-        return filedata
+        return self._get_remote_file(self, "pending")
```
Comment (JordiMForgeFlow) [inline edi_storage_oca/components/receive.py]:
```suggestion
        return self._get_remote_file("pending")
```

File: edi_storage_oca/components/listener.py [hunk 4]
```diff
@@ -17,14 +19,15 @@ def _move_file(self, storage, from_dir_str, to_dir_str, filename):
...
-        files = storage.list_files(from_dir.as_posix())
+        files = utils.list_files(storage, from_dir.as_posix())
         files = [os.path.basename(f) for f in files]
         if filename not in files:
             return False
-        # TODO: clean this up, .move_files is deprecated in fs_storage
         self._add_post_commit_hook(
-            storage.move_files, [(from_dir / filename).as_posix()], to_dir.as_posix()
+            utils.move_files,
+            self.storage,
```
Comment (JordiMForgeFlow) [inline edi_storage_oca/components/listener.py]:
```suggestion
            storage,
```

Comment (JordiMForgeFlow) [reply edi_storage_oca/components/listener.py]: _add_post_commit_hook should also be modified to receive the storage parameter

File: edi_storage_oca/utils.py (hunk 1 above)
Comment (sebalix) [reply edi_storage_oca/utils.py]: I was working on this as this triggered this error during our tests: `AttributeError: 'str' object has no attribute 'as_posix'`.
I removed all calls to `as_posix` in these helpers so they now receive only paths as strings. Should they accept only `pathlib.Path` objects instead? :man_shrugging:

File: edi_storage_oca/components/receive.py [hunk 5]
```diff
@@ -14,7 +14,4 @@ class EDIStorageReceiveComponent(Component):
     _usage = "storage.receive"
 
     def receive(self):
-        path = self._get_remote_file_path("pending")
-        # TODO: clean this up, .get is deprecated in fs_storage
-        filedata = self.storage.get(path.as_posix())
-        return filedata
+        return self._get_remote_file("pending")
```
Comment (JordiMForgeFlow) [inline edi_storage_oca/components/receive.py]:
```suggestion
        return self._get_remote_file("pending", binary=True)
```

**Review by JordiMForgeFlow — APPROVED** [review]
Code + functional review 👍🏼

File: edi_storage_oca/utils.py [hunk 6]
```diff
@@ -0,0 +1,60 @@
+# Copyright 2024 Camptocamp SA
+# License LGPL-3.0 or later (http://www.gnu.org/licenses/lgpl)
+
+import base64
+import os
+import re
+
+
+def add_file(storage, path, filedata, binary=False):
```
Comment (simahawk) [inline edi_storage_oca/utils.py:9]: nice. This is something that should probably be proposed to the base module as I don't see any edi specific stuff.

**Review by simahawk — APPROVED** [review]

**Comment by simahawk** [comment]
@sebalix can you squash pls?

File: edi_storage_oca/utils.py (hunk 6 above)
Comment (sebalix) [reply edi_storage_oca/utils.py:9]: It was in the base module `fs_storage` before (as Odoo methods, not functions), but deprecated. Thing is we still need to encapsulate such features somewhere yes.

**Pull request merged**